syntax/odoc.vim

```diff
@@ -0,0 +1,82 @@
+" Vim syntax file
+" Language:     Odoc/OCamldoc
+" Filenames:    *.mld
+" Maintainers:  Samuel Hym
+"               Nicolas Osborne
+" URL:          https://github.com/ocaml/vim-ocaml
+" Last Change:
+"               2024 Jan 26 - initial version
+
+" Quit when a syntax file was already loaded
+if !exists("odoc_syntax_loading")
+  if exists("b:current_syntax")
+    finish
+  endif
+  let odoc_syntax_loading = 1
+endif
+
+let s:keepcpo = &cpo
+set cpo&vim
+
+syn case match
+
+syn include @odocSyntaxOCaml syntax/ocaml.vim
+unlet b:current_syntax
+
+syn cluster odocInline contains=odocBold,odocItalic,odocEmphasis,odocMiscInline,odocList,odocLink,odocCode,odocCrossref,odocCodeBlock,odocVerbatim,odocTargetSpecific,odocTag
+
+syn region odocLink matchgroup=odocLinkMarker start="{" end="}" contains=odocCrossref
+syn region odocBold matchgroup=odocMarker start="{b\>" end="}" contains=@Spell,@odocInline
+syn region odocEmphasis matchgroup=odocMarker start="{e\>" end="}" contains=@Spell,@odocInline
+syn region odocItalic matchgroup=odocMarker start="{i\>" end="}" contains=@Spell,@odocInline
+syn region odocMiscInline matchgroup=odocMarker start="{[CLR^_]" end="}" contains=@Spell,@odocInline
+syn region odocVerbatim matchgroup=odocMarker start="{v\>" end="\<v}"
+
+if get(g:,'odoc_html_support',0)
+  syn include @odocHtml syntax/html.vim
+  unlet b:current_syntax
+  syn region odocTargetSpecific matchgroup=odocMarker start="{%html:" end="%}" contains=@odocHtml
+else
+  syn region odocTargetSpecific matchgroup=odocMarker start="{%html:" end="%}" contains=@Spell
+endif
+
+syn region odocDyckWord contained start="{" end="}" contains=odocDyckWord
+syn region odocTargetSpecific matchgroup=odocMarker start="{m\%(ath\)\?\>" end="}" contains=odocDyckWord
+syn region odocTargetSpecific matchgroup=odocMarker start="{%\%(latex\|texi\|man\):" end="%}" contains=@Spell
+syn region odocHeading start="{[0-5]" end="}" contains=@Spell
+syn region odocTable matchgroup=odocMarker start="{t\>" end="}" contains=@Spell,@odocInline
+syn region odocTable matchgroup=odocMarker start="{table\>" end="}" contains=odocTableRow
+syn region odocTableRow matchgroup=odocMarker start="{tr\>"  end="}" contains=odocTableEntry
+syn region odocTableEntry matchgroup=odocMarker start="{t[dh]\>"  end="}" contains=@Spell,@odocInline
+syn region odocCode matchgroup=odocMarker start="\[" end="\]"
```

Comment:
A last bug I just noticed:
[inline code spans have special escape rules.][man-code-escape] They can contain escaped square brackets, i.e. `[ \[ ]` and `[ \] ]`, which are rendered as `[` and `]` respectively (no other character are escaped, not even backslashes, so `[ \\ \\[ ]` is rendered as `\\ \[`). They can also contain *balanced* unescaped brackets, like in this example:

    The call [transpose [[1;2;3];[4;5;6]]] returns [[[1;4];[2;5];[3;6]]].

(This only concerns square brackets. There is nothing special about braces, parentheses or quotes. Moreover, code blocks opened by `{v` or `{[` support no escaping at all.)

Add this to support this syntax:

    syn match odocEscapedBracket contained "\\[][]"
    syn region odocBalancedBracket contained transparent start="\[" end="\]"
    " later (`odocEscaped` comes from another reviewing comment):
    hi def link odocEscapedBracket odocEscaped

and add `contains=odocBalancedBracket,odocEscapedBracket` to the `odocCode` region (there is no need to add it to `odocBalancedBracket` region, because its `transparent` attribute already takes care of it, see `:help syn-transparent`).

I find this is a supplementary argument for highlighting the contents of code spans a bit more visibly.

[man-code-escape]: https://ocaml.github.io/odoc/odoc_for_authors.html#escaping